scripts/build_failure_data_collection.py

```diff
@@ -0,0 +1,294 @@
+        "include_fields": fields,
+        "resolution": "---",
+        "f1": "creation_ts",
+        "o1": "greaterthan",
+        "v1": one_year_ago,
+        "f2": "longdesc",
+        "o2": "allwordssubstr",
+        "v2": "backed out for causing build",
+    }
+    return params
+
+
+def get_backed_out_build_failure_bugs(date="today", bug_ids=[], chunk_size=None):
+    params = get_bz_params()
+    bugs = {}
+
+    def bug_handler(bug, data):
+        data[bug["id"]] = bug
+
+    Bugzilla(
+        params,
+        bughandler=bug_handler,
+        bugdata=bugs,
+    ).get_data().wait()
+
+    return bugs
+
+
+def map_bugs_to_commit(bug_ids):
+    logger.info("Mapping bugs to their commits...")
+    bug_commits = {}
+
+    for commit in tqdm(
+        repository.get_commits(
+            include_no_bug=True, include_backouts=True, include_ignored=True
+        )
+    ):
+        if commit["bug_id"] not in bug_ids:
+            continue
+
+        commit_data = {
+            key: commit[key]
+            for key in ["node", "bug_id", "pushdate", "backedoutby", "backsout", "desc"]
+        }
+
+        bug_commits.setdefault(commit["bug_id"], []).append(commit_data)
+
+    return bug_commits
+
+
+def find_bugs(hg_client, bug_ids, bug_commits):
+    logger.info("Finding bugs...")
+    backed_out_revisions = []
+
+    for bug_id in bug_ids:
+        bug_id_commits = bug_commits.get(bug_id, None)
+        backing_out_commit = find_backing_out_commit(bug_id_commits, hg_client)
+
+        if not backing_out_commit:
+            continue
+
+        logger.info("Backing out commit found!")
+
+        # commit = {}
+
+        # commit["desc"] = next(
+        #     (
+        #         c["desc"]
+        #         for c in bug_id_commits
+        #         if any(
+        #             c["node"].startswith(node)
+        #             for node in backing_out_commit["backsout"]
+        #         )
+        #     ),
+        #     None,
+        # )
+
+        # if commit["desc"] is None:
+        #     continue
+
+        commits = [
+            {
+                "desc": c["desc"],
+            }
+            for c in bug_id_commits
+            if any(
+                c["node"].startswith(node) for node in backing_out_commit["backsout"]
+            )
+        ]
+
+        if commits is None:
+            continue
+
+        for commit in commits:
+            revision_id = repository.get_revision_id(commit)
+            backed_out_revisions.append(revision_id)
+
+    return backed_out_revisions
+
+
+def find_backing_out_commit(commits, hg_client):
+    logger.info("Finding backing out commit...")
+    if not commits:
+        return None
+
+    for commit in commits:
+        if not commit["backsout"]:
+            continue
+
+        desc = commit["desc"]
+        if (
+            "backed out" in desc.lower()
+            and "for causing" in desc.lower()
+            and "build" in desc.lower()
+        ):
+            return commit
```

Comment:
in this case why do you need bugzilla data?

Reply:
I thought commits via hg had metadata:
https://hg.mozilla.org/mozilla-central/rev/99a8f2b2b00d85148c743f16db75d9abefb33513

if you want a short list of commits that are backed out, then bugzilla comment parsing would do.

One value add method would be to propose a change to the backout process where there is queryable metadata so everytime there is a backout, data is stored in bugzilla/treeherder/some_random_db with original commit, backed out commit, related links, list of failing tasks and related error messages.  Thinking more, developers relanding would have to go through a process to document what they were relanding, and confirm the interdiff :)

Reply:
my original comment here was assuming you were doing a massive hg log to look at commit history and parse out "backed out" commit messages to build a list.